components/markets/MarketScroll.tsx

```diff
@@ -1,38 +1,48 @@
-import { motion } from "framer-motion";
-import { useStore } from "lib/stores/Store";
 import { observer } from "mobx-react";
-import Link from "next/link";
 import { useEffect, useRef, useState } from "react";
-import { ChevronLeft, ChevronRight } from "react-feather";
 import { useResizeDetector } from "react-resize-detector";
-import MarketCard, { IndexedMarketCardData } from "./market-card";
+import MarketCard, { IndexedMarketCardData } from "./market-card/index";
+import HorizontalScroll from "components/ui/HorizontalScroll";
 
 const MarketScroll = observer(
   ({
     title,
     markets,
-    showMarketsLink = true,
+    link,
   }: {
     title: string;
     markets: IndexedMarketCardData[];
-    showMarketsLink?: boolean;
+    link?: string;
   }) => {
     const scrollRef = useRef<HTMLDivElement>();
     const [scrollLeft, setScrollLeft] = useState(0);
     const [scrollDirection, setScrollDirection] = useState<"left" | "right">(
       "right",
     );
     const { width: containerWidth, ref: containerRef } = useResizeDetector();
-    const cardWidth = 320;
-    const gap = 30;
+    const gap = 16;
+    //calculate cards shown and width based on container width
+    const cardsShown =
+      containerWidth < 716
+        ? 3
+        : containerWidth >= 716 && containerWidth < 1183
+        ? 2
+        : 3;
+    // const cardsShown = Math.floor(containerWidth / (gap + cardWidth));
+    const cardWidth =
+      containerWidth < 716
+        ? containerWidth
+        : containerWidth >= 1183
+        ? (containerWidth - gap * 2) / cardsShown
+        : (containerWidth - gap) / cardsShown;
     const scrollMin = 0;
     const scrollMax = cardWidth * markets.length + gap * (markets.length - 1);
-    const cardsShown = Math.floor(containerWidth / (gap + cardWidth));
+
     const moveSize = cardsShown * (cardWidth + gap);
 
     useEffect(() => {
       scrollRef.current.scroll({ left: scrollLeft, behavior: "smooth" });
-    }, [scrollRef, scrollLeft]);
+    }, [scrollRef, scrollLeft, cardWidth]);
```

Comment:
strange behavior in my opinion, when the screen width (and the cardWidth) is changed, the component scrolls the market cards. On smallest breakpoint cards can end up looking like this
![image](https://user-images.githubusercontent.com/551965/213145441-29bc9e6c-0c93-40b3-a035-8c08ff0435ed.png)

Reply:
Yes, actually upon review adding cardWidth to the useEffect is unnecessary. Width prop gets updating just fine without it.